pkg/kubectl/cmd/config/create_authinfo.go

```diff
@@ -95,17 +112,55 @@ func (o createAuthInfoOptions) run() error {
 func (o *createAuthInfoOptions) modifyAuthInfo(existingAuthInfo clientcmdapi.AuthInfo) clientcmdapi.AuthInfo {
 	modifiedAuthInfo := existingAuthInfo
 
+	var setToken, setBasic, setCert bool
+
 	if o.authPath.Provided() {
 		modifiedAuthInfo.AuthPath = o.authPath.Value()
 	}
+
 	if o.clientCertificate.Provided() {
 		modifiedAuthInfo.ClientCertificate = o.clientCertificate.Value()
+		if len(modifiedAuthInfo.ClientCertificate) > 0 {
+			modifiedAuthInfo.ClientCertificateData = nil
+			setCert = true
+		}
 	}
 	if o.clientKey.Provided() {
 		modifiedAuthInfo.ClientKey = o.clientKey.Value()
+		if len(modifiedAuthInfo.ClientKey) > 0 {
+			modifiedAuthInfo.ClientKeyData = nil
+		}
 	}
+
 	if o.token.Provided() {
 		modifiedAuthInfo.Token = o.token.Value()
+		setToken = len(modifiedAuthInfo.Token) > 0
+	}
+
+	if o.username.Provided() {
+		modifiedAuthInfo.Username = o.username.Value()
+		setBasic = setBasic || len(modifiedAuthInfo.Username) > 0
+	}
+	if o.password.Provided() {
+		modifiedAuthInfo.Password = o.password.Value()
+		setBasic = setBasic || len(modifiedAuthInfo.Password) > 0
+	}
+
+	// If any auth info was set, make sure any other existing auth types are cleared
```

Comment:
Would be better to throw an error if mutually exclusive auth types were provided. If the user's input is invalid, it's less surprising to fail early with an actionable message than silently ignore values.